### ISSUE TITLE
Fix visual bugs for dark mode

### ISSUE THREAD

**Issue by tracytees** [Shopify/polaris-react-archive#7986]
<img width="1310" alt="Screenshot 2023-01-04 at 2 02 27 PM" src="https://user-images.githubusercontent.com/108495662/210666744-710c24e8-6d35-4c38-bbbf-2ba57a80ed20.png">
Noticed that the new pattern pages aren't dark mode friendly. As of right now, we aren't removing dark mode for the Polaris site. 

**Dark mode accessibility checklist:**

- [x] Visual examples aren't overly contrasted in dark mode (i.e not pure white or pure black backgrounds)
- [x] Tabs and text are readable
- [x] Ensure colours used are not oversaturated
- [x] If changing tab system background colour, perhaps consider shadows that work for both modes
- [ ] Treatment decision should be added into style guide

**Comment by SeanyB** [comment]
Great catch. I certainly think we should ensure that patterns appear correctly at all times in the Polaris website, but I think we should be careful about how much time we invest in supporting Dark Mode. If this isn't straightforward to address, we should discuss why.

**Comment by johanstromqvist** [comment]
Gave this a quick look.

### Observations and takeaways

<details>
<summary>
Let's consider visual imperfections a site level problem, and out of scope for patterns v1.</summary>

#### Example from Spacing guidelines
![Image](https://user-images.githubusercontent.com/38329124/210864847-54740e92-6202-47f3-b7d4-6800043664c5.png)

#### Not pretty, but not broken

![Image](https://user-images.githubusercontent.com/38329124/210872781-dc4039f9-0a69-4627-ad8e-1294ef36aa10.png)


</details>
<details>
<summary>
Some issues have critical impact on functionality and accessibility. Let's fix those asap.
</summary>

#### Issue 1
The majority of the issues seem to be caused by the background and border of the tabbed content. I wonder if we could resolve most of those by using similar transparency based styling as the props table.

![Image](https://user-images.githubusercontent.com/38329124/210870954-c159ea5d-f4c2-451e-8f7a-c2644acb2ea5.png)

![Image](https://user-images.githubusercontent.com/38329124/210872326-94517d24-0e72-4a03-9d37-330a4927445f.png)

Current styling
![Image](https://user-images.githubusercontent.com/38329124/210870360-5d01eaaf-4c16-492c-8f46-b9d3bb40f4bd.png)

Props table styling
![Image](https://user-images.githubusercontent.com/38329124/210870853-9baba284-d0d1-43e7-8136-690d83996c09.png)

#### Issue 2

This issue is caused by the text switching styling when toggling dark mode, but the background remains white.
![Image](https://user-images.githubusercontent.com/38329124/210869028-2cf7d827-5fbf-445e-833e-1036759795db.png)


</details>
<details>
<summary>
Some visual issues are caused by our unique styling in Patterns. Let's fix the quick wins.
</summary>


#### Tables on the site

![Image](https://user-images.githubusercontent.com/38329124/210866562-1502ce1b-7381-45dd-84db-8eecb8fb4023.png)

![Image](https://user-images.githubusercontent.com/38329124/210868614-7a16b952-58e5-4837-8118-c9ab50b6f202.png)

#### Pattern page tables
![Image](https://user-images.githubusercontent.com/38329124/210866626-6d9a4086-d5d7-4e3a-a610-03c540b16f1f.png)

</details>

### In summary
- [x] Adjust table borders to match [prop table](https://polaris.shopify.com/components/alpha-stack) and [other tables](https://polaris.shopify.com/design/space)
- [x] Fix background color and border color of tabbed content (perhaps same bg and border as tables)
- [x] Change the text color of the code template action bar to the link color of "Edit in CodeSandbox" on component page

Escalate to site level
- [x] Audit and fix dark mode imperfections

Does that seem about right? Seen anything else?

**Comment by SeanyB** [comment]
Thanks for this summary, @johanstromqvist. I agree, let's focus on the critical issues specific to Patterns in this project.

**Comment by tracytees** [comment]
I've made some [adjustments to the design](https://www.figma.com/file/JUKo15oKz1aCcdeNP4UxyE/Reboot-Polaris-Patterns?node-id=5065%3A260861&t=ktWL5OwhM0LAxecC-1) based on the feedback. Would love some eyes @raquelbreternitz (since you're in charge of the visuals) 
At the moment, table borders are white and are overly contrasted/distracting if we change the tab background outline to white @johanstromqvist. 
I've made some annotations to the design as well but happy to answer any questions or concerns.

**Comment by johanstromqvist** [comment]
Thanks @tracytees! Could you list the changes made please, so that it's easy to find and check them?

A few observations:
- The tab borders and background look good. Are they new colors or reusing existing colors from the site?
- There's no "when to use" table in the mockup. Is that intentional?
- The current text-link color on the site is `#8bb2ff`. It would be ideal to reuse that one, assuming it meets a11y standards on the slightly lighter tab background.
- Links in "related resources" look like they still have light mode colors. They would be the same text-link color as above.

> At the moment, table borders are white and are overly contrasted/distracting if we change the tab background outline to white

I was referring to the existing tables on the site, e.g. the props table on the [component page](https://polaris.shopify.com/components/alpha-stack). Ideally we can reuse the same colors for similar purposes on the site, and the colors that are already in use are likely thoughtfully designed from an a11y POV.

**Comment by tracytees** [comment]
Thanks for the feedback. Here is a list of the adjustments I've made:

- Tab BG is now `#1E1E23 `with stroke `#373737 `to reuse/match what we currently have on the site
- Text link colour is `#8BB2FF` to re-use the existing link treatment for dark mode
- Table was re-added and the background for the header and table body is `#1E1E23` 
Take a look at the [Figma file](https://www.figma.com/file/JUKo15oKz1aCcdeNP4UxyE/Reboot-Polaris-Patterns?node-id=5065%3A260861&t=ktWL5OwhM0LAxecC-1) if you'd like

**Comment by raquelbreternitz** [comment]
Mostly LGTM, except for **some minor contrast issues**

- The purple numbers on the white squares fail AA (bringing the opacity of the background square up to 95% fixes this) 
- The gray links that are the `subdued text` color (breadcrumbs, Github link) fail AA. They're fine if we adjust them to match the breadcrumbs on the site, which according to the inspect tool is `rgba(128, 128, 137)` (might be good to double-check the actual site code) versus our subdued token, which is `rgba(109, 113, 117, 1)`.

**A few other thoughts**:
- I noticed that tags on the App Design Guidelines and "beta/alpha" in the nav swap out color+opacity for dark mode, we may want to do similarly for our component tags like like `date picker` and  `text` 
- The tab stroke outline feels VERY light to me, I can barely see it on my high-quality retina screen, but if that matches the rest of the site then not worth changing for now. However, it also, it looks like it's brighter within the "use when..." table, I assume because it's on an opacity and doubled? (that might just be a figma thing)
- As an FYI, if we match the rest of the site for dark mode table styling, the title row is bold+lighter text and has a slightly lighter background, you can see an example in the [Contributing](https://polaris.shopify.com/contributing) section: 
<img width="658" alt="Screenshot 2023-01-25 at 12 44 52 PM" src="https://user-images.githubusercontent.com/13204374/214654934-5a3e1e3e-46d4-4b6c-99ba-247cf303c7e5.png">


<sub><sup>P.S. honestly I think it'd be great for us to consider adapting image assets to dark mode but that's a huge project for the entirety of the docs and will probably never feel like high enough priority until the admin itself is available in a dark mode 😅, besides maybe being something we just want to do as a total brand overhaul of the site, so not worth tackling for now.</sub></sup>

**Comment by johanstromqvist** [comment]
Nice catches!

Perhaps the fixes that require reusable site-wide styles could be summarized in a single task like "Make sure element X, Y, Z matches existing site styles" or something? Cus what we want at the end of the day is that common pattern elements matches common component elements (breadcrumbs, text links, tables, etc) while other styling issues that are more unique to the pattern page might need "new" design decisions.

**Comment by raquelbreternitz** [comment]
Good point. Summarized:

Match to existing dark mode styling:
- [x] Tags
- [x] Table (title row)
- [x] Subdued links (breadcrumbs, github)

Adjust to be AA:
- [x] Increase opacity of the white bg on the annotations to 95%

**Comment by johanstromqvist** [comment]
Nice!

When this design task is done, please move it forward by:
- summarize the required fixes
- remove the `design` label
- move to `this week`
- notify eng on Slack

**Comment by tracytees** [comment]
Changes made:

- Background changed to `#F0F0F0` for number tags under `How it helps merchants`, which now complies with AA accessibility
- Subdued links changed to `#85858E` which is a new colour...the existing one (`#808089`) fails accessibility. Would suggest `text-subdued` to be changed globally to `#85858E`
- Verified border stroke colour to be `#373737`
- Background for entire tab section changed to `#18181D`, which is a new colour (the background should stand out slightly but open to changing to back to regular surface colour (`#161619)
- Table header background colour changed to `#1E1E23` and text should be `--text-strong: #e7e7f1` (how it is elsewhere)
- To match ADG style: Tags background changed to `#002852` (`--color-docs-text-link-highlight`) 
- To match ADG style: Tags font and border colour changed to `#b5c9fa` (`--color-docs-text-link-default` from ADG)

**Comment by gwyneplaine** [comment]
^changes in the following PR #8261, just need a lil design love for the tab hover styles noted in [this PR comment](https://github.com/Shopify/polaris/pull/8261#issuecomment-1415052547)

**Comment by SeanyB** [comment]
This looks good! Thank you!
